pkg/controller/dbsyncer/generic_db_to_transport_syncer.go

```diff
@@ -81,28 +76,70 @@ func (syncer *genericDBToTransportSyncer) initLastUpdateTimestampFromTransport()
 }
 
 func (syncer *genericDBToTransportSyncer) syncBundle(ctx context.Context) {
-	lastUpdateTimestamp, err := syncer.db.GetLastUpdateTimestamp(ctx, syncer.dbTableName)
-	if err != nil {
-		syncer.log.Error(err, "unable to sync bundle to leaf hubs", syncer.dbTableName)
-		return
-	}
+	ticker := time.NewTicker(syncer.intervalPolicy.GetInterval())
 
-	if !lastUpdateTimestamp.After(*syncer.lastUpdateTimestamp) { // sync only if something has changed
-		return
+	for {
+		select {
+		case <-ctx.Done(): // we have received a signal to stop
+			ticker.Stop()
+			return
+
+		case <-ticker.C:
+			lastUpdateTimestamp, err := syncer.db.GetLastUpdateTimestamp(ctx, syncer.dbTableName)
+			if err != nil {
+				syncer.completeSync(ticker, false)
+				syncer.log.Error(err, "unable to sync bundle to leaf hubs", syncer.dbTableName)
+
+				continue
+			}
+
+			if !lastUpdateTimestamp.After(*syncer.lastUpdateTimestamp) { // sync only if something has changed
+				syncer.completeSync(ticker, false)
+
+				continue
+			}
+
+			// if we got here, then the last update timestamp from db is after what we have in memory.
+			// this means something has changed in db, syncing to transport.
+			bundleResult := syncer.createBundleFunc()
+			lastUpdateTimestamp, err = syncer.db.GetBundle(ctx, syncer.dbTableName, syncer.createObjFunc, bundleResult)
+
+			if err != nil {
+				syncer.completeSync(ticker, false)
+				syncer.log.Error(err, "unable to sync bundle to leaf hubs", syncer.dbTableName)
+
+				continue
+			}
+
+			syncer.lastUpdateTimestamp = lastUpdateTimestamp
+
+			syncer.syncToTransport(syncer.transportBundleKey, datatypes.SpecBundle, lastUpdateTimestamp, bundleResult)
+			syncer.completeSync(ticker, true)
+		}
 	}
-	// if we got here, then the last update timestamp from db is after what we have in memory.
-	// this means something has changed in db, syncing to transport.
-	bundleResult := syncer.createBundleFunc()
-	lastUpdateTimestamp, err = syncer.db.GetBundle(ctx, syncer.dbTableName, syncer.createObjFunc, bundleResult)
+}
 
-	if err != nil {
-		syncer.log.Error(err, "unable to sync bundle to leaf hubs", syncer.dbTableName)
-		return
+// completeSync notifies policy whether sync was actually performed or skipped and resets ticker's interval
+// to a new recalculated one.
+func (syncer *genericDBToTransportSyncer) completeSync(ticker *time.Ticker, syncPerformed bool) {
```

Comment:
I think this function has logic of two different things and should be changed.
one aspect of this function (the main one) is to re-evaluate the sync interval (check if it was changed by the policy).
this logic should probably stay here but function name should be changed accordingly (something like like `evaluateInterval` or `reevaluateInterval`).

the other thing that is done here and I think should be removed is this code:
```
if syncPerformed {
	syncer.intervalPolicy.OnSyncPerformed()
} else {
	syncer.intervalPolicy.OnSyncSkipped()
}
```

this part is not related to the reevaluation of the interval based on the policy and should be done outside of this function.

Reply:
done